framework/src/vectorpostprocessors/PositionsFunctorValueSampler.C

```diff
@@ -0,0 +1,93 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PositionsFunctorValueSampler.h"
+#include "Positions.h"
+#include "FEProblemBase.h"
+
+registerMooseObject("MooseApp", PositionsFunctorValueSampler);
+
+InputParameters
+PositionsFunctorValueSampler::validParams()
+{
+  InputParameters params = PointSamplerBase::validParams();
+  params.addClassDescription(
+      "Sample one or more functors at points specified by a Positions object.");
+  params.addRequiredParam<PositionsName>(
+      "positions",
+      "Positions object specifying the points where you want to evaluate the functors");
+  params.addRequiredParam<std::vector<MooseFunctorName>>(
+      "functors", "The functors we want to evaluate at the positions");
+  params.addParam<bool>("discontinuous", false, "Indicate that the functors are discontinuous");
+
+  return params;
+}
+
+PositionsFunctorValueSampler::PositionsFunctorValueSampler(const InputParameters & parameters)
+  : PointSamplerBase(parameters),
+    NonADFunctorInterface(this),
+    _positions(_fe_problem.getPositionsObject(getParam<PositionsName>("positions")))
+{
+  const auto & functor_names = getParam<std::vector<MooseFunctorName>>("functors");
+  for (const auto & functor_name : functor_names)
+    _functors.push_back(&getFunctor<Real>(functor_name));
+
+  _discontinuous_at_faces = getParam<bool>("discontinuous");
+
+  // Initialize the data structures in SamplerBase
+  std::vector<std::string> functor_string_names;
+  for (const auto & functor_name : functor_names)
+    functor_string_names.push_back(functor_name);
+  SamplerBase::setupVariables(functor_string_names);
+}
+
+void
+PositionsFunctorValueSampler::initialize()
+{
+  // Pull new points
+  _points = _positions.getPositions(_fe_problem.getCurrentExecuteOnFlag() == EXEC_INITIAL);
+
+  PointSamplerBase::initialize();
+}
+
+void
+PositionsFunctorValueSampler::execute()
+{
+  BoundingBox bbox = _mesh.getInflatedProcessorBoundingBox();
+  const auto state = determineState();
+
+  // Pull new points
+  _points = _positions.getPositions(_fe_problem.getCurrentExecuteOnFlag() == EXEC_INITIAL);
+
+  for (const auto i : index_range(_points))
+  {
+    Point & p = _points[i];
+
+    // Do a bounding box check so we're not doing unnecessary PointLocator lookups
```

Comment:
I would also pull in the second line of the corresponding comment in `PointVariableSamplerBase.C`. Also does `contains_point` return `true` if the point is on the boundary of the box? If so, I don't really understand having the second half of this `||`

Reply:
there is no guarantee on contains_point, it s an absolute check
https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1BoundingBox.html#aabb3e5cbc3b7bb5f7e99ddfacc6fdc08

I ll move the rest of the comment

Reply:
actually since the BBOX has an inflation factor we should be fine removing it everywhere

Reply:
Let s try it. Turning on distributed recipes

Reply:
Cant do that. I remember why now

Added more comments